Realm/Realm/DatabaseTypes/RealmDictionary.cs

```diff
@@ -98,6 +98,18 @@ public ICollection<TValue> Values
             }
         }
 
+        internal RealmResults<TValue> ToResults()
+        {
+            return (RealmResults<TValue>)Values;
+        }
+
+        // Get filtered results from dictionary's values
+        internal RealmResults<TValue> GetFilteredValueResults(string query, RealmValue[] arguments)
```

Comment:
I think moving it here and renaming it slightly was most straightforward way to make this dictionary specific and not confusing in terms of overriding `GetFilteredResults` from base class but perhaps there's better naming/ approach.